.github/workflows/main_win.yaml

```diff
@@ -187,6 +187,43 @@ jobs:
           type: onhost
           os: windows
 
+  # Runs on-device integration and unit tests.
+  on-device-test:
+    needs: [initialize, build]
+    # Run ODT when on_device label is applied on PR.
+    # Also, run ODT on push and schedule if not explicitly disabled via repo vars.
+    if: |
+      fromJSON(needs.initialize.outputs.on_device_test).enabled == true && ((
+        github.event_name == 'pull_request' &&
+        contains(github.event.pull_request.labels.*.name, 'on_device') ) || ((
+            inputs.nightly == 'true' || github.event_name == 'schedule') &&
+            vars.RUN_ODT_TESTS_ON_NIGHTLY != 'False') ||
+          ( github.event_name == 'push' && vars.RUN_ODT_TESTS_ON_POSTSUBMIT != 'False' ) )
+    runs-on: [self-hosted, odt-runner]
+    name: ${{ matrix.name }}_on_device_${{ matrix.shard }}
+    container: ${{ needs.docker-unittest-image.outputs.docker_unittest_tag }}
+    permissions: {}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ${{ fromJson(needs.initialize.outputs.platforms) }}
+        config: [devel]
+        shard: ${{ fromJson(needs.initialize.outputs.on_device_test).tests }}
+        include: ${{ fromJson(needs.initialize.outputs.includes) }}
+    env:
+      COBALT_BOOTLOADER: ${{ needs.initialize.outputs.bootloader }}
+      ON_DEVICE_TEST_ATTEMPTS: ${{ needs.initialize.outputs.on_device_test_attempts }}
```

Comment:
And here's yet another one. `on_device_test_attempts` isn't defined in the initialize step.